graphql_compiler/schema/__init__.py

```diff
@@ -301,26 +301,33 @@ def _parse_date_value(value):
     return arrow.get(value, "YYYY-MM-DD").date()
 
 
-def _serialize_datetime(value):
+def _serialize_datetime(value: Any) -> str:
     """Serialize a DateTime object to its proper ISO-8601 representation."""
     # Python datetime.datetime is a subclass of datetime.date, but in this case, the two are not
     # interchangeable. Rather than using isinstance, we will therefore check for exact type
     # equality.
-    if type(value) not in {datetime, arrow.Arrow}:
+    #
+    # We don't allow Arrow objects as input since it seems that Arrow objects are always tz aware.
+    # This is supported by the fact that the `.naive` Arrow method returns a datetime object instead
+    # of an Arrow object.
+    if type(value) == datetime and value.tzinfo is None:
+        return value.isoformat()
+    else:
         raise ValueError(
-            "Expected argument to be a python datetime object. "
-            "Got {} of type {} instead.".format(value, type(value))
+            f"Expected a timezone naive datetime object. Got {value} of type {type(value)} instead."
         )
-    return value.isoformat()
 
 
-def _parse_datetime_value(value):
+def _parse_datetime_value(value: Any) -> datetime:
```

Comment:
Can we say `value` here is a string?

Reply:
Changing the type annotation to `str` does not give us a mypy type error, but I think that it might make more sense to have `Any` here and in `_serialize_datetime`

GraphQL core next has an `Any` type annotation for the input values of `parse_value` and `serialize` functions and GraphQLJS has a `mixed` type annotation for these values, which seems to be the JS equivalent of `Any`. 
https://flow.org/en/docs/types/mixed/


I get the feeling that this more likely going to lead to unnecessary `coerce_type` calls than to prevent any bugs. What do you think? 

A specific case I am thinking about is what if we had:

```
def deserialize_json_value(json_value: Union[str, int, ...], expected_type: GraphQLType):
      ....
```
Since you can't pass  a `Union[str, ...]` argument into a function with a `str` type annotation, we would probably have to have a lot of `coerce_type` internally within this function. 

I am thinking of a similar scenario for the `serialize` function. We use the `serialize` value for validation. In the future, we might update the `validate_argument_type` type annotation to:

```
ALLOWED_PYTHON_TYPES: Union[str, int, ...]

def validate_argument_type(
       name: str, 
       value: ALLOWED_PYTHON_TYPES, 
       graphql_type: GraphQLType
)  -> None:
```

For a similar reason, we would have to a lot of `coerce_type` calls internally to make this function pass mypy linting.